### PR TITLE
Parallelize Windows Builds

### PR DESCRIPTION
**Pull request by MariaSemple**
This PR changes our make program from `nmake` to `ninja` in our windows builds to allow use to compile multiple files in parallel. It also sets the Windows SDK version to ensure we're building with the desired SDK version (based on the Dockerfile).

### CODE REVIEW

File: docker/jenkins/Dockerfile.windows [hunk 1]
```diff
@@ -22,11 +22,12 @@ RUN choco install -y cmake --installargs 'ADD_CMAKE_TO_PATH=""System""' --fail-o
   choco install -y -i ant; `
   choco install -y windows-sdk-10.1 --version 10.1.17134.12; `
   choco install -y 7zip; `
-  choco install -y nsis
-  
+  choco install -y nsis; `
+  choco install -y ninja --verion "Ninja 1.10.0"
```
Comment (kevinushey) [inline docker/jenkins/Dockerfile.windows:26]:
```suggestion
  choco install -y ninja --version "Ninja 1.10.0"
```

File: package/win32/make-install-win32.bat [hunk 2]
```diff
@@ -18,14 +18,17 @@ cd %WIN32_BUILD_PATH%
...
+cmake -G "Ninja" ^
       -DCMAKE_INSTALL_PREFIX:String=%INSTALL_PATH% ^
       -DCMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE% ^
       -DRSTUDIO_TARGET=SessionWin32 ^
       -DRSTUDIO_PACKAGE_BUILD=1 ^
+      -DCMAKE_C_COMPILER=cl.exe ^
+      -DCMAKE_CXX_COMPILER=cl.exe ^
       ..\..\.. || goto :error
-cmake --build . --config %CMAKE_BUILD_TYPE% --target install || goto :error
+cmake --build . %MAKEFLAGS% --config %CMAKE_BUILD_TYPE% --target install || goto :error
```
Comment (kevinushey) [inline package/win32/make-install-win32.bat:31]:
```suggestion
cmake --build . --config %CMAKE_BUILD_TYPE% --target install -- %MAKEFLAGS% || goto :error
```

IIRC, CMake usually expects flags that will be passed to the underlying build tool at the end, following `--`.

File: package/win32/make-package.bat [hunk 3]
```diff
@@ -39,21 +39,22 @@ if exist "%BUILD_DIR%/_CPack_Packages" rmdir /s /q "%BUILD_DIR%\_CPack_Packages"
...
+call "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\BuildTools\Common7\Tools\VsDevCmd.bat" -arch=amd64 -startdir=none -host_arch=amd64 -winsdk=10.0.17134.0 -no_logo || goto :error
+cmake -G "Ninja" ^
       -DRSTUDIO_TARGET=Desktop ^
       -DCMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE% ^
       -DRSTUDIO_PACKAGE_BUILD=1 ^
+      -DCMAKE_C_COMPILER=cl ^
+      -DCMAKE_CXX_COMPILER=cl ^
       ..\..\.. || goto :error
-cmake --build . --config %CMAKE_BUILD_TYPE% || goto :error
+cmake --build . %MAKEFLAGS% --config %CMAKE_BUILD_TYPE% || goto :error
```
Comment (kevinushey) [inline package/win32/make-package.bat:51]:
```suggestion
cmake --build . --config %CMAKE_BUILD_TYPE% -- %MAKEFLAGS% || goto :error
```

**Comment by gtritchie** [comment]
> Do we need to do anything to ensure that `ninja` is on the PATH for development environments? (It looks like `ninja.exe` is normally bundled as part of the Visual Studio installation; perhaps it gets put on the PATH by the `VsDevCmd.bat` invocations?)

`ninja` is also installed by dev machine bootstrapper: https://github.com/rstudio/rstudio/blob/master/dependencies/windows/Install-RStudio-Prereqs.ps1 script

**Pull request merged**